### DISCUSSION TITLE
HW 2 Part 3.2

### DISCUSSION THREAD

**Discussion by Chello28** [cooperrc/computational-mechanics#90] [General]
Confused on the error message received in code when trying to make a linear regression to find constants  𝐴  and  𝐵  for our Moore's law fit.
![image](https://user-images.githubusercontent.com/70422314/190467291-db69f41a-8ce4-4aac-8ff3-c1d91665cc92.png)

**Comment by cooperrc** [comment]
The `RankWarning` is telling you that the `polyfit` solution is poorly defined. This _usually_ comes up when you're trying to fit higher order polynomials to small data sets. I can force the same error, if I use 1 (x, y) data points e.g. 

```python
x = 0
y = 3
np.polyfit(x, y, 1)

out: /opt/conda/lib/python3.9/site-packages/IPython/core/interactiveshell.py:3441: RankWarning: Polyfit may be poorly conditioned
  exec(code_obj, self.user_global_ns, self.user_ns)
```

My guess: `xi` and `yi` are only length 1 arrays somehow
